File: app/src/main/java/com/lipeng/mygithub/homepage/presenter/HomePagePresenter.java
package com.lipeng.mygithub.homepage.presenter;

/**
 * 主页面的presenter
 * @author lipeng
 * @date 2018/1/3
 */

public interface HomePagePresenter {
    /**
     * 点击RecyclerView子项跳转至项目详情页面
     * */
    void skipToProjectDetail();

    /**
     * 销毁资源
     * */
    void destroy();
}

File: app/src/main/java/com/lipeng/mygithub/ui/presenter/HomePagePresenter.kt
package com.lipeng.mygithub.ui.presenter

import com.lipeng.mygithub.app.AppData
import com.lipeng.mygithub.base.MvpBasePresenter
import com.lipeng.mygithub.ui.contract.IHomePageContract

/**
 * @author big insect
 */
class HomePagePresenter : MvpBasePresenter<IHomePageContract.IHomePageView>() ,IHomePageContract.IHomePagePresenter {

    override fun logout() {
        daoSession.authUserDao.delete(AppData.authUser)
        AppData.authUser = null
        AppData.loggedUser = null
        if (isViewAttached){
            view.restartApp()
        }
    }
}